src/App.tsx

```diff
@@ -31,12 +31,32 @@ declare global {
 }
 
 const tele: any = window.Telegram.WebApp;
+const allowedPublicIP = "96.9.70.228";
 
 function App() {
   const [cartItems, setCartItems] = useState<CartItem[]>([]);
 
   useEffect(() => {
     tele.ready();
+
+    const checkUserIP = async () => {
+      try {
+        const response = await fetch("https://api.ipify.org?format=json");
+        const data = await response.json();
+        const userPublicIP = data.ip;
+
+        if (userPublicIP !== allowedPublicIP) {
+          document.body.innerHTML = "<h1>Access Denied: You must be connected to the abc-wifi network.</h1>";
```

Comment:
Should we declare another variable of <h1> as a new const?

Reply:
`const errrMsg = `<h1>Access Denied: You must be connected to the abc-wifi network.</h1>`